CLUE_BBQ/clue_bbq.py

```diff
@@ -0,0 +1,148 @@
+    my_labels[name] = templabel
+    text_group.append(templabel)
+
+temperatures_screen.append(text_group)
+
+temp_title_label = label.Label(title_font, text="BBQLUE", color=clue.ORANGE, max_glyphs=15)
+temp_title_label.x = 12
+temp_title_label.y = 30
+temperatures_screen.append(temp_title_label)
+
+# PyLint can't find BLERadio for some reason so special case it here.
+ble = adafruit_ble.BLERadio()  # pylint: disable=no-member
+
+ibbq_connection = None
+
+while True:
+    # re-display homescreen here
+    clue.display.show(homescreen_screen)
+
+    print("Scanning...")
+    for adv in ble.start_scan(ProvideServicesAdvertisement, timeout=5):
+        clue.pixel.fill((50, 50, 0))
+        if IBBQService in adv.services:
+            print("found an IBBq advertisement")
+            ibbq_connection = ble.connect(adv)
+            print("Connected")
+            break
+
+    # Stop scanning whether or not we are connected.
+    ble.stop_scan()
+
+    try:
+        if ibbq_connection and ibbq_connection.connected:
+            clue.pixel.fill((0, 0, 50))
+            ibbq_service = ibbq_connection[IBBQService]
+            ibbq_service.init()
+            while ibbq_connection.connected:
+
+                if clue.button_a:  # hold a to swap between C and F
+                    print("unit_mode swapped")
+                    unit_mode = not unit_mode
+                    clue.red_led = True
+                    clue.play_tone(1200, 0.1)
+                    clue.red_led = False
+                    time.sleep(0.1)  # debounce
+
+                temps = ibbq_service.temperatures
+                batt = ibbq_service.battery_level
+                if temps is not None:
+                    probe_count = len(temps)  # check how many probes there are
+                    for i in range(probe_count):
+                        if (temps[i] is not 0 and temps[i] < 1000):  # unplugged probes
+                            if unit_mode:
+                                clue.pixel.fill((50, 0, 0))
+                                temp = temps[i]
+                                my_labels[i].text = "{} C".format(temp)
+                                clue.pixel.fill((0, 0, 0))
+                                print("Probe", i+1, "Temperature:", temp,"C",)
+                            else:  # F
+                                clue.pixel.fill((50, 0, 0))
+                                temp = temps[i] * 9 / 5 + 32
+                                my_labels[i].text = "{} F".format(temp)
+                                clue.pixel.fill((0, 0, 0))
+                                print("Probe", i+1, "Temperature:", temp,"F",)
+                        else:
+                            print("Probe", i+1, "is unplugged",)
+                            my_labels[i].text = ("  ---")
+                    clue.display.show(temperatures_screen)
+
+    except _bleio.ConnectionError:
+        # Redisplay splash screen here, or put it at top of while True:
```

Comment:
Remove this comment; it was just a note to you in the sample code I sent you.

Reply:
gone